### DISCUSSION TITLE
Your input on Hyas

### DISCUSSION THREAD

**Discussion by h-enk** [gethyas/hyas#239] [Ideas]
What would you like to see added to Hyas? Share your input in a comment below. Thanks!

**Comment by bip7686** [comment]
A partial of "render-image.html" would be nice!

**Reply by h-enk** [reply]
Thanks for your input — noted.

**Reply by h-enk** [reply]
https://github.com/h-enk/hyas/issues/260

**Reply by h-enk** [reply]
@bip7686, now part of [Hyas v1.5.6](https://github.com/h-enk/hyas/releases/tag/v1.5.6) — see also the [image partial](https://github.com/h-enk/hyas/blob/master/layouts/partials/content/image.html), the [settings](https://github.com/h-enk/hyas/blob/master/config/_default/params.toml#L48-L50), and the [documentation](https://gethyas.com/docs/recipes/images/)

**Reply by bip7686** [reply]
Awesome, thank you!

**Comment by 7underlines** [comment]
Green icon at lighthouse pwa test.

**Reply by h-enk** [reply]
Added https://github.com/h-enk/hyas/issues/284

**Comment by spookyuser** [comment]
Just wanna say, I found this repo while searching for a way to minify scripts only on prod - using cs.github.com - and the way you have done the esbuild.html file is genius! Thank you sooooo much 🥰

**Reply by h-enk** [reply]
Totally agree! All credits to @sean-au

**Comment by pensivedog** [comment]
Easy themeability! Ideally, to be able to use [theme components](https://gohugo.io/hugo-modules/theme-components/) to manage customizations in `themes/my-custom-styles-and-hyas-overrides` while keeping `themes/hyas` synced with upstream improvements. Perhaps this could somehow be achieved by adding hyas as a git submodule to a standard Hugo installation?

I know the topic of themeability has come up in separate issues and discussion threads... just adding it here since you asked for what we want to see added to Hyas. Thank you for all your efforts and keep up the great work!

**Reply by sean-au** [reply]
It would probably be easier to use Hyas as a theme via hugo modules, rather than getting involved with git submodules. 

The prerequisite is hyas being initialised as a hugo module.

**Reply by h-enk** [reply]
Will be looking into this (again/with fresh eyes + a couple of years later). I think that Git Submodules still is a no-go (see also the references below) + I suspect that Hugo Modules still have some blocking limitations (compared to npm modules). I also think it's not smart to manage dependencies using both Hugo Modules + npm modules, So, I suspect it will come down to using a modular setup using npm modules.

To be continued.

Resources:
- [Using Git Submodules Effectively](https://www.philosophicalhacker.com/post/using-git-submodules-effectively/)
- [Master Hugo Modules: Managing Themes As Modules](https://www.hugofordevelopers.com/articles/master-hugo-modules-managing-themes-as-modules/)

**Reply by h-enk** [reply]
See also: https://github.com/gethyas/hyas/issues/313#issuecomment-1596589293
